### PR TITLE
Use json editor

### PR DESCRIPTION
**Pull request by cyrexin**
This PR adds the json editor component. #129  

### Description of changes
User is now able to edit the json (ontology & config) in a json editor (https://github.com/josdejong/jsoneditor) instead of plain text.

### Possible influences of this PR
The json editor component can be used for other json if needed.

### Demonstration of results
json_editor

### CODE REVIEW

**Review by hunterhector — COMMENTED** [review]
Interesting. This looks cool. Any ideas on travis failing?

File: src/app/components/jsonEditor/JsonEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  componentDidUpdate() {
+    this.jsoneditor.updateText(this.props.jsonText);
+  }
+
+  render() {
+    return (
+        <div className={`jsoneditor-react-container ${ this.props.className }`} ref={this.container} />
+    );
+  }
+}
```
[the remark this answers is not in the document]
Comment (cyrexin) [reply src/app/components/jsonEditor/JsonEditor.tsx]: Done

File: src/app/components/jsonEditor/index.tsx [hunk 2]
```diff
@@ -0,0 +1 @@
+export { default } from './JsonEditor';
```
[the remark this answers is not in the document]
Comment (cyrexin) [reply src/app/components/jsonEditor/index.tsx]: Done

**Comment by cyrexin** [comment]
> Interesting. This looks cool. Any ideas on travis failing?

Fix the travis test by the guidance here: https://jestjs.io/docs/en/webpack#mocking-css-modules

**Review by hunterhector — COMMENTED** [review]
This seems to work well. If we have a JSON Schema for this, how can we add it to the JsonEditor component?

**Comment by cyrexin** [comment]
> This seems to work well. If we have a JSON Schema for this, how can we add it to the JsonEditor component?

As the example shows [here](https://github.com/josdejong/jsoneditor/blob/develop/examples/07_json_schema_validation.html), you can provide the schema as a param and the jsoneditor will do the validation.

**Comment by hunterhector** [comment]
> > This seems to work well. If we have a JSON Schema for this, how can we add it to the JsonEditor component?
> 
> As the example shows [here](https://github.com/josdejong/jsoneditor/blob/develop/examples/07_json_schema_validation.html), you can provide the schema as a param and the jsoneditor will do the validation.

Shall we make this parameter available at the component level?

https://github.com/asyml/stave/pull/137/files#diff-9ede13f0aee5572e8933d3ab41a8b6fc00270d6b179e6c482370c2882fa87253R16

Here, the `options` are specified and take the `onChangeText` function. Shall we make the option available to the outside caller, so we can easily add the `schema` option when using it?

**Review by hunterhector — CHANGES_REQUESTED** [review]
Let's expose the schema to the higher level caller either by exposing the whole option or just that parameter. I think we can make use of the schema very soon.

Btw, if the json fails the schema check, what would happen?

**Review by hunterhector — APPROVED** [review]

**Pull request merged**